winnowing.py

```diff
@@ -3,122 +3,118 @@
 import hashlib
 from cleanUP import *
 
-#sha-1 encoding is used to generate hash values
-def hash(text):
-    #this function generates hash values
-    hashval = hashlib.sha1(text.encode('utf-8'))
-    hashval = hashval.hexdigest()[-4 :]
-    hashval = int(hashval, 16)  #using last 16 bits of sha-1 digest
-    return hashval
+class winnowing():
+    def __init__(self,file1,file2):
+        self.file1=file1
+        self.file2=file2
+        self.token1 = tokenize(file1)  #from cleanUP.py
+        self.str1 = toText(self.token1)
+        self.token2 = tokenize(file2)
+        self.str2 = toText(self.token2)
+        self.kGrams1 = self.kgrams(self.str1)  #stores k-grams, their hash values and positions in cleaned up text
+        self.kGrams2 = self.kgrams(self.str2)
+        self.HL1 = self.hashList(self.kGrams1)  #hash list derived from k-grams list
+        self.HL2 = self.hashList(self.kGrams2)
+        self.fpList1 = self.fingerprints(self.HL1)
+        self.fpList2 = self.fingerprints(self.HL2)
 
-#function to form k-grams out of the cleaned up text
-def kgrams(text, k = 25):
-    tokenList = list(text)
-    n = len(tokenList)
-    kgrams = []
-    for i in range(n - k + 1):
-        kgram = ''.join(tokenList[i : i + k])
-        hv = hash(kgram)
-        kgrams.append((kgram, hv, i, i + k))  #k-gram, its hash value, starting and ending positions are stored
-        #these help in marking the plagiarized content in the original code.
-    return kgrams
-
-#function that returns the index at which minimum value of a given list (window) is located
-def minIndex(arr):
-    minI = 0
-    minV = arr[0]
-    n = len(arr)
-    for i in range(n):
-        if arr[i] < minV:
-            minV = arr[i]
-            minI = i
-    return minI
-
-#we form windows of hash values and use min-hash to limit the number of fingerprints
-def fingerprints(arr, winSize = 4):
-    arrLen = len(arr)
-    prevMin = 0
-    currMin = 0
-    windows = []
-    fingerprintList = []
-    for i in range(arrLen - winSize):
-        win = arr[i: i + winSize]  #forming windows
-        windows.append(win)
-        currMin = i + minIndex(win)
-        if not currMin == prevMin:  #min value of window is stored only if it is not the same as min value of prev window
-            fingerprintList.append(arr[currMin])  #reduces the number of fingerprints while maintaining guarantee
-            prevMin = currMin  #refer to density of winnowing and guarantee threshold (Stanford paper)
-
-    return fingerprintList
+    def plagiarismCheck(self):
+        file1 = self.file1
+        file2 = self.file2
+        f1 = open(file1, "r")
```

Comment:
This needs to be changed. Instead of passing the file path, send the file contents. Check this [link](https://docs.python.org/3/library/io.html) to store a string in memory file object.

Reply:
sir we need path in cleanUP.py file . if we send only the content of file as a string ,we have to make some changes in cleanUP.py file as well . Should I do this ?

Reply:
Yes you need to change. The changes I mentioned here (getting list elements) are also required in cleanUP.py file.